### PR TITLE
Fix screen size detection

### PR DESCRIPTION
**Pull request by Jamminroot**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR will fix screen size detection, and report it as seen in preferences ('Looks like...').

## Related PRs or issues

[AB#80463](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/80463)

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m [hunk 1]
```diff
@@ -374,21 +374,28 @@ - (NSNumber *)timeZoneOffset:(NSTimeZone *)timeZone {
...
-  CGFloat scale = 3;
-  CGSize screenSize = [UIScreen mainScreen].bounds.size;
+  id screen = [NSClassFromString(@"NSScreen") valueForKey:@"mainScreen"];
+  if (screen == nil) {
+    CGSize screenSize = [UIScreen mainScreen].nativeBounds.size;
+    return [NSString stringWithFormat:@"%dx%d", (int)(screenSize.width), (int)(screenSize.height)];
+  }
+  NSInvocation *invocation =
+      [NSInvocation invocationWithMethodSignature:[[screen class] instanceMethodSignatureForSelector:NSSelectorFromString(@"frame")]];
+  [invocation setSelector:NSSelectorFromString(@"frame")];
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m:388]: Reuse the selector

File: AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m [hunk 2]
```diff
@@ -374,21 +374,28 @@ - (NSNumber *)timeZoneOffset:(NSTimeZone *)timeZone {
...
+  return [NSString stringWithFormat:@"%dx%d", (int)screenSize.width, (int)screenSize.height];
 #elif TARGET_OS_MACCATALYST
-  // This returns scale value based on scale values iOS.
-  // So we need to use base value for macOS.
-  CGFloat scale = 3;
-  CGSize screenSize = [UIScreen mainScreen].bounds.size;
+  id screen = [NSClassFromString(@"NSScreen") valueForKey:@"mainScreen"];
+  if (screen == nil) {
+    CGSize screenSize = [UIScreen mainScreen].nativeBounds.size;
+    return [NSString stringWithFormat:@"%dx%d", (int)(screenSize.width), (int)(screenSize.height)];
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m:384]:
```suggestion
    return [NSString stringWithFormat:@"%dx%d", (int)screenSize.width, (int)screenSize.height];
```

File: AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m [hunk 3]
```diff
@@ -374,21 +374,29 @@ - (NSNumber *)timeZoneOffset:(NSTimeZone *)timeZone {
 }
 
 - (NSString *)screenSize {
-
 #if TARGET_OS_OSX
-  NSScreen *focusScreen = [NSScreen mainScreen];
-  CGFloat scale = focusScreen.backingScaleFactor;
-  CGSize screenSize = [focusScreen frame].size;
+  NSSize screenSize = [NSScreen mainScreen].frame.size;
+  return [NSString stringWithFormat:@"%dx%d", (int)screenSize.width, (int)screenSize.height];
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m:379]: Please add comments about "Looks like" right to the code

File: AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m [hunk 4]
```diff
@@ -374,21 +374,31 @@ - (NSNumber *)timeZoneOffset:(NSTimeZone *)timeZone {
...
+  }
+  SEL selector = NSSelectorFromString(@"frame");
+  NSInvocation *invocation =
+      [NSInvocation invocationWithMethodSignature:[[screen class] instanceMethodSignatureForSelector:selector]];
+  [invocation setSelector:selector];
+  [invocation setTarget:screen];
+  [invocation invoke];
+  CGRect frame;
+  [invocation getReturnValue:&frame];
+  return [NSString stringWithFormat:@"%dx%d", (int)frame.size.width, (int)frame.size.height];
```
Comment (russelarms) [inline AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m:396]: For macos and maccatalyst it shows "screenSize" : "1440x900" (actual is "2880x1800"), while for ios it is "1334x750" (actual is "1334x750").
If this approach for macos &maccatalyst is showing correct numbers why we can't multiply them by the scale factor as we did before? Does it have any caveats?
I think we should handle it following the same logic for all the platforms.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,13 @@
 # App Center SDK for iOS, macOS and tvOS Change Log
 
+## Version 3.3.0 (Under development)
+
+### App Center
+
+* **[Fix]** Fix screen resolution detection and order of width and height fields for different platforms.
```
Comment (MatkovIvan) [inline CHANGELOG.md:7]: > for different platforms

Need to be more precise here

File: AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m [hunk 6]
```diff
@@ -374,21 +374,37 @@ - (NSNumber *)timeZoneOffset:(NSTimeZone *)timeZone {
...
 - (NSString *)screenSize {
-
 #if TARGET_OS_OSX
-  NSScreen *focusScreen = [NSScreen mainScreen];
-  CGFloat scale = focusScreen.backingScaleFactor;
-  CGSize screenSize = [focusScreen frame].size;
+  
+  // This would report screen resolution as shown in display settings ('Looks like' field in scaling tab).
+  // 'Scale' value reflects the scale factor needed to convert from the default logical
+  // coordinate space into the device coordinate space of this screen.
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m:381]: The last change is not correct

**Pull request merged**